rust/zypp-agama/src/lib.rs

```diff
@@ -66,12 +68,39 @@ where
     Some(progress_callback::<F>)
 }
 
-pub fn init_target<F>(root: &str, progress: F) -> Result<(), ZyppError>
+// Mutex for the Zypp C API which is not thread safe.
+// TODO: guard some (global) context instead of ().
```

Comment:
That's my main question: which is exactly the purpose of this mutex? What we are protecting the access to? Or do you want to avoid having more than one instance of Zypp?

Reply:
Yes, currently libzypp can only have one instance, via C++ `zypp::getZYpp`, but our current Rust API does not reflect that.
So I made a `Zypp` Rust object, guarded by a `Mutex` to have only one instance.
In C-layer we have `zypp_ptr()` as a function but do not have an exposed data type for it, therefore in Rust I have nothing to wrap and use `()` as the "contents" of the Mutex. It is technically wrong, but good enough. I'll check if introducing a Zypp type at the c-layer level is easy+useful enough.